iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/GenericUDFIcebergBucket.java

```diff
@@ -158,6 +159,16 @@ public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumen
         };
         break;
 
+      case DATE:
+        converter = new PrimitiveObjectInspectorConverter.DateConverter(argumentOI,
+          PrimitiveObjectInspectorFactory.writableDateObjectInspector);
+        Transform<Integer, Integer> dateTransform = Transforms.bucket(Types.DateType.get(), numBuckets);
+        evaluator = arg -> {
+          DateWritableV2 val = (DateWritableV2) converter.convert(arg.get());
+          result.set(dateTransform.apply(val.get().toEpochDay()));
```

Comment:
How about `val.getDays()` instead of `val.get.toEpochDay()`?

Reply:
👍